### PR TITLE
added bar chart

### PR DESCRIPTION
**Pull request by Kieran-Fishwick-TfN**
Describe Changes
---
added bar chart to plot functionality

Needs
 - main merging to get latest commits
 - docstrings
 - tests
 
Task Checklist
----
- [ ] Have unittests been added (testing individual functions and their edge cases)
- [ ] Have integration tests been added (if applicable, to test modules of functionality)
- [ ] Updated RELEASE.md to reflect changes in PR
- [ ] Have new dependencies been added?
  - [ ] Have they been added to either `requirements.txt` or `requirements_dev.txt`.
  - [ ] Have they been added to `pyproject.toml`

### CODE REVIEW

File: examples/hist_plot_examples.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import matplotlib.pyplot as plt
```
Comment (MattBuckley-TfN) [inline examples/hist_plot_examples.py:1]: Please could you add a title and description for the example, probably with some links back to the function used and then separate the example file into code block with some basic explanation of the chunk which generates the data and the bit which produces the plot.

File: src/caf/viz/xy_plot.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
 # Third Party
 import matplotlib as mpl
 import numpy as np
+import pandas as pd
```
Comment (MattBuckley-TfN) [inline src/caf/viz/xy_plot.py:16]: Please could you ignore the ruff rule about type checking block and explain why

File: src/caf/viz/xy_plot.py [hunk 3]
```diff
@@ -228,6 +230,17 @@ def scatter(
...
 
+def bar(
+    fig: figure.Figure,
+    ax: axes.Axes,
+    data: BasicData,
+) -> None:
+    ax.bar(data.data[data.x_column], data.data[data.y_column])
+    ax.set_xlabel(data.x_label)
+    ax.set_ylabel(data.y_label)
+    ax.set_title(data.title)
```
Comment (MattBuckley-TfN) [inline src/caf/viz/xy_plot.py:241]: These need some checks so they're only ran if the data attribute is not None

File: src/caf/viz/xy_plot.py [hunk 4]
```diff
@@ -228,6 +230,17 @@ def scatter(
         fig.colorbar(points, ax=ax, label="Density", ticks=ticker.NullLocator())
 
 
+def bar(
```
Comment (MattBuckley-TfN) [inline src/caf/viz/xy_plot.py:233]: Missing docstring

File: src/caf/viz/xy_plot.py [hunk 5]
```diff
@@ -228,6 +230,17 @@ def scatter(
...
+    fig: figure.Figure,
```
Comment (MattBuckley-TfN) [inline src/caf/viz/xy_plot.py:234]: Ignore ruff comment and explain why

**Review by MattBuckley-TfN — CHANGES_REQUESTED** [review]
Please could you fix the linting / typing issues and also the docs building is failing for some reason.

Also, could you add some more detail to the example, explaining how it works.
